as/cNFT/assembly/models/persistent_tokens_royalty.ts

```diff
@@ -14,10 +14,20 @@ export function defaultTokenRoyalty(): TokenRoyalty {
     }
 }
 
+@nearBindgen
+export class BidShares {
+    constructor(
+        public prev_owner: u16,
+        public creator: u16,
+        public owner: u16
+    ) {}
+}
+
 @nearBindgen
 export class PeristentTokenRoyalty {
     /** @todo explain this structure*/
     private _trmap: PersistentUnorderedMap<TokenId, TokenRoyalty>
+    private _tshmap: PersistentUnorderedMap<TokenId, BidShares>
```

Comment:
Sorry if this was not clear but the idea here is not to add BidShares here but to integrate it's function, which is similar to `TokenRoyalty` inside `_trmap` and this `PersistentTokenRoyalty`. I'm going to try writing more details in the issues.